modules/oauth/src/main/java/org/mule/runtime/module/oauth2/internal/authorizationcode/AuthorizationCodeGrantType.java

```diff
@@ -19,9 +20,24 @@
 public interface AuthorizationCodeGrantType extends ApplicationCredentials {
 
   /**
-   * @return redirect url as defined in the oauth authentication server.
+   * @return callback listener configuration to start the server and define corresponding flow.
    */
-  String getRedirectionUrl();
+  HttpListenerConfig getLocalCallbackConfig();
+
+  /**
+   * @return callback listener path for the flow to be created according to localCallbackConfig.
+   */
+  String getLocalCallbackConfigPath();
+
+  /**
+   * @return local callback URL for the flow to be created.
+   */
+  String getLocalCallbackUrl();
+
+  /**
+   * @return redirect url to be sent to the client, won't start a server there.
```

Comment:
The last comment seems odd, can it be something like it's the way the server is seen externally?